### PR TITLE
Add Cling dev GSOC proposal

### PR DESCRIPTION
**Pull request by oshadura**
cc: @vgvassilev

### CODE REVIEW

File: _gsocproposals/2021/proposal_cling_dev.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * Static code analysis (clang-tidy)
+ * Coding conventions checks (clang-format)
+ * Release binary upload automation
+
+The student should be prepared to write a progress report and present the results.
+
+## Necessary skills
+Understanding of CMake; GitHub; GitHub actions; codecov; clang-tidy and clang-format
+
+## Mentors
```
Comment (aflp91) [inline _gsocproposals/2021/proposal_cling_dev.md:29]: You need to add mentors in `gsoc/2021/mentors.md` file. (sorted in **last name** alphabetic order)

And, please, put the main mentor in bold letters there, in the proposal.

File: _gsocproposals/2021/proposal_cling_dev.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+* [Oksana Shadura](mailto:Oksana.shadura@cern.ch)
+* [Vassil Vassilev](mailto:vvasilev@cern.ch)
+
+## Links
```
Comment (aflp91) [inline _gsocproposals/2021/proposal_cling_dev.md:33]: I think this list would be nicer if organized as a list (with `*` in front of each item) …

**Review by aflp91 — CHANGES_REQUESTED** [review]

File: _gsocproposals/2021/proposal_cling_dev.md (hunk 1 above)
Comment (oshadura) [reply _gsocproposals/2021/proposal_cling_dev.md:33]: Done!

**Review by agheata — APPROVED** [review]

File: _gsocproposals/2021/proposal_cling_dev.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
+title: Improve Cling’s Development Lifecycle
```
Comment (agheata) [inline _gsocproposals/2021/proposal_cling_dev.md:2]: There is already a proposal having this name, see: https://hepsoftwarefoundation.org/gsoc/2021/proposal_cling_cuda.html

**Review by agheata — CHANGES_REQUESTED** [review]

File: _gsocproposals/2021/proposal_cling_dev.md (hunk 3 above)
Comment (vgvassilev) [reply _gsocproposals/2021/proposal_cling_dev.md:2]: Doh, I copy-pasted the wrong title from the google doc. Good catch!!

Comment (oshadura) [reply _gsocproposals/2021/proposal_cling_dev.md:2]: @agheata should I add a fix in this PR for another proposal? (here the title is correct)

Comment (agheata) [reply _gsocproposals/2021/proposal_cling_dev.md:2]: In the other proposal pls, we merge this after the fix

Comment (oshadura) [reply _gsocproposals/2021/proposal_cling_dev.md:2]: Please check #930

**Review by agheata — APPROVED** [review]

**Pull request merged**